api/v1alpha5/virtualmachine_storage_types.go

```diff
@@ -354,25 +354,40 @@ type VirtualMachineStorageStatus struct {
 	Used *VirtualMachineStorageStatusUsed `json:"usage,omitempty"`
 }
 
+type VirtualMachineStorageStatusUsedSnapshotDetails struct {
```

Comment:
The snapshot should have an `Other` category, no?

Reply:
Yeah we discussed this and thought this is not a P0. The Snapshot detail can just be categorized into
- VM: space used by VirtualMachineSnapshot (which includes other + disks).
- Volume: space used by VolumeSnapshot.

Reply:
Discussed this offline.
Summary:
Andrew is ok with current code if it doesn't impact how we report snapshot usage to quota framework.
We don't use this data to report quota.